// next-prefixed/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Link = exports.Image = exports.prefixURL = void 0;

var _react = _interopRequireDefault(require("react"));

var _link = _interopRequireDefault(require("next/link"));

var _config = _interopRequireDefault(require("next/config"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

var _getConfig = (0, _config.default)(),
    publicRuntimeConfig = _getConfig.publicRuntimeConfig;

var prefixPath = function prefixPath(path) {
  return publicRuntimeConfig.assetPrefix.replace(/\/+$/, '') + '/' + (path || '').replace(/^\/+/, '');
};

var prefixURL = function prefixURL(url) {
  if (_typeof(url) === 'object') {
    return Object.assign({}, url, {
      pathname: prefixPath(url.pathname)
    });
  }

  return /^(?:[a-z][a-z0-9+.-]*:|\/\/)/i.test(url) ? url : prefixPath(url);
};

exports.prefixURL = prefixURL;

var Image = function Image(props) {
  return _react.default.createElement("img", _extends({}, props, {
    src: prefixURL(props.src)
  }));
};

exports.Image = Image;

var Link = function Link(props) {
  return _react.default.createElement(_link.default, _extends({}, props, {
    as: prefixURL(props.as || props.to || props.href)
  }));
};

exports.Link = Link;
